lang/rustup/Makefile

```diff
@@ -0,0 +1,64 @@
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+include $(TOPDIR)/rules.mk
+
+PKG_NAME:=rustup
+PKG_VERSION:=1.22.1
+PKG_RELEASE:=1
+PKG_LICENSE:=MIT
+
+PKG_SOURCE_PROTO:=git
+PKG_SOURCE_URL:=https://github.com/rust-lang/rustup.git
+PKG_SOURCE_DATE:=2020-07-18
+PKG_SOURCE_VERSION:=ac1a8c415b5c2886275ca92a3a8f3a4cd0587711
+PKG_HASH:=46c1cf800e48f43bf0b3e19bf1eb2ed414106e5ff2647cf7858286489454acf2
+PKG_BUILD_DIR:=$(BUILD_DIR)/$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE_SUBDIR:=$(PKG_NAME)-$(PKG_VERSION)
+PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
+
+include $(INCLUDE_DIR)/host-build.mk
+include $(INCLUDE_DIR)/package.mk
+include ./rustc-triple.mk
+
+define Package/rustup
+    SECTION:=lang
+    CATEGORY:=Languages
+    TITLE:=Rust Programming Language Installer
+    URL:=https://www.rust-lang.org/
+endef
+
+define Package/rustup/description
+ The rustup package installs the Rust Programming Language (rustc/cargo)
+ for use with packages that require Rust to compile.
+
+ This is a HOST ONLY package; no installed files to the target device.
+endef
+
+define Host/Configure
+	RUSTUP_HOME=$(HOST_RUST_DIR) \
+	CARGO_HOME=$(HOST_CARGO_DIR) \
+	$(BUILD_DIR_HOST)/$(PKG_SOURCE_SUBDIR)/rustup-init.sh -y -v --profile default --target $(RUSTC_TARGET_ARCH)
+endef
+
+define Host/Install
+	true
+endef
+
+define Host/Compile
+	true
+endef
+
+define Build/InstallDev
+	RUSTUP_HOME=$(HOST_RUST_DIR) CARGO_HOME=$(HOST_CARGO_DIR) $(HOST_CARGO_DIR)/bin/rustup target add $(RUSTC_TARGET_ARCH)
+endef
```

Comment:
I don't think you can rely on `Build/InstallDev` to be called before rustc is used by a package build, e.g. if host tools are built, then the target is changed with `make menuconfig`, if this (rustup) package is not selected (and it should not be), then `Build/InstallDev` would not be run.

I think it would be better to add a pre-build hook (either `Hooks/Configure/Pre` or `Hooks/Build/Pre`) in rustc-triple.mk to add the target if necessary.

Reply:
the other packages that use rustup would need a PKG_HOST_DEPENDS/PKG_BUILD_DEPENDS for rustup/host.  Would that cause a race against Build/InstallDev?

Reply:
`Build/InstallDev` is part of the target package build (for this package). `PKG_BUILD_DEPENDS=rustup/host` is host package build. (There is no `PKG_HOST_DEPENDS`.) Host and target package builds take place independently of each other (unless the target package depends on the host package).